File: app/src/main/java/com/absinthe/anywhere_/model/QREntity.java
package com.absinthe.anywhere_.model;

import com.absinthe.anywhere_.interfaces.OnQRLaunchedListener;

public class QREntity {
    private String mPkgName;
    private String mClsName;
    private String mUrlScheme;
    private OnQRLaunchedListener mListener;

    QREntity(OnQRLaunchedListener listener) {
        this.mListener = listener;
    }

    public void launch() {
        mListener.onLaunched();
    }

    public String getPkgName() {
        return mPkgName;
    }

    public void setPkgName(String mPkgName) {
        this.mPkgName = mPkgName;
    }

    public String getClsName() {
        return mClsName;
    }

    public void setClsName(String mClsName) {
        this.mClsName = mClsName;
    }

    public String getUrlScheme() {
        return mUrlScheme;
    }

    public void setUrlScheme(String mUrlScheme) {
        this.mUrlScheme = mUrlScheme;
    }

    public OnQRLaunchedListener getListener() {
        return mListener;
    }

    public void setListener(OnQRLaunchedListener mListener) {
        this.mListener = mListener;
    }
}

File: app/src/main/java/com/absinthe/anywhere_/model/QREntity.kt
package com.absinthe.anywhere_.model

import com.absinthe.anywhere_.interfaces.OnQRLaunchedListener

class QREntity internal constructor(var listener: OnQRLaunchedListener) {

    var pkgName: String? = null
    var clsName: String? = null
    var urlScheme: String? = null

    fun launch() {
        listener.onLaunched()
    }

    override fun toString(): String {
        return "QREntity: pkgName = $pkgName, clsName = $clsName, urlScheme = $urlScheme"
    }
}